### ISSUE TITLE
"method not implemented" when using RoomRead.getMembers(roomId)

### ISSUE THREAD

**Issue by sharpner** [RocketChat/Rocket.Chat.Apps-engine#75]
Hi, 

we are trying create an app that creates a room name dynamically via the usernames, but within the executePreRoomCreateModify we are not able to figure out the usernames. 

Is there a different way with an implemented method? Or any other idea on how to fix this? 
:)

thanks in advance

**Comment by graywolf336** [comment]
Sadly, there is currently no way to get users which are in a room. :( Rodrigo is currently working on implementing items to allow getting users in a room.

**Comment by sharpner** [comment]
Thanks for the fast response! 

I see.... Well at least we know now that it is planned, so there is something to look forward for :)

**Comment by graywolf336** [comment]
Yeah, one of my Apps broke in the latest updates (Out of Office) due to the usernames removal, so I feel the pain.

**Comment by sharpner** [comment]
Oh that's unfortunate. Kinda the basics of a chat though, users and messages :/ But great to hear that someone is working on it! Is there any issue I can subscribe to?

**Comment by graywolf336** [comment]
Currently there's no issue or pull request to watch, however there is a branch: https://github.com/RocketChat/Rocket.Chat.Apps-engine/tree/usernames-getter which this issue is a good one to watch.

**Comment by graywolf336** [comment]
But it does fix the `room.usernames`. They used to not have the usernames of the users in the rooms, but it does now.

**Comment by winterstefan** [comment]
Hi @graywolf336, thank you very much for the changes and keeping us up-to-date!

> we are trying create an app that creates a room name dynamically via the usernames, but within the executePreRoomCreateModify we are not able to figure out the usernames.

I tried to update the app @sharpner mentioned.

1) After upgrading Rocket.Chat to `0.71.0`, our app detail page stated '_Failed to fetch_'.
2) After setting the `requiredApiVersion` in the `app.json` from `^0.9.13` (found in several existing apps) to `^1.2.1` (Is that correct?), I still got '_Failed to fetch_'.

Taking a deeper look, the `AppApiManager. listApis()` resulted in `No app found for the provided id.`.


Could you give us a hint what might be wrong?

**Comment by graywolf336** [comment]
1. Is the App actually installed?
2. If so, is it activated?
3. Can you connect with me on our Open Community server, that way we can investigate further..

**Comment by winterstefan** [comment]
1. Yes 👍 
  - It _was_ installed just before I upgraded Rocket.Chat. Then, after having the '_Failed to fetch_', I humbly tried to remove the app by dropping it from the `rocketchat_app_*` mongo collections (uninstall via UI wasn't possible).
  - Then, after a second install (now with the updated `requiredApiVersion`), the same error appeared.
2. Correct, it was activated in both cases.
3. I'll investigate my issue, once again, and come back to you asap - connecting with you via https://open.rocket.chat.

**Comment by graywolf336** [comment]
@rodrigok or @d-gubert any ideas on how to achieve the above comment?

**Comment by sharpner** [comment]
the behaviour described by @winterstefan is still happening in v1.3.2. 
Any updates on this issue?
